File: packages/react-native-vision-camera-location/ios/HybridLocation.swift
///
/// HybridLocation.swift
/// VisionCameraLocation
/// Copyright © 2025 Marc Rousavy @ Margelo
///

import CoreLocation
import VisionCamera

final class HybridLocation: HybridLocationSpec, NativeLocation {
  let location: CLLocation

  var latitude: Double {
    return location.coordinate.latitude
  }
  var longitude: Double {
    return location.coordinate.longitude
  }
  var altitude: Double {
    return location.altitude
  }

  var horizontalAccuracy: Double {
    return location.horizontalAccuracy
  }

  var verticalAccuracy: Double {
    return location.verticalAccuracy
  }

  var timestamp: Double {
    let unixSeconds = location.timestamp.timeIntervalSince1970
    return unixSeconds * 1000.0
  }

  var isMock: Bool {
    if let source = location.sourceInformation {
      return source.isSimulatedBySoftware
    }
    return false
  }

  init(location: CLLocation) {
    self.location = location
    super.init()
  }
}

File: packages/react-native-vision-camera-location/android/src/main/java/com/margelo/nitro/camera/location/HybridLocation.kt
package com.margelo.nitro.camera.location

import android.location.Location
import android.os.Build
import com.margelo.nitro.camera.HybridLocationSpec
import com.margelo.nitro.camera.public.NativeLocation

class HybridLocation(
  override val location: Location,
) : HybridLocationSpec(),
  NativeLocation {
  override val latitude: Double
    get() = location.latitude
  override val longitude: Double
    get() = location.longitude
  override val altitude: Double
    get() = location.altitude
  override val horizontalAccuracy: Double
    get() = location.accuracy.toDouble()
  override val verticalAccuracy: Double
    get() =
      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
        location.verticalAccuracyMeters.toDouble()
      } else {
        -1.0
      }
  override val timestamp: Double
    get() = location.time.toDouble()
  override val isMock: Boolean
    get() =
      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
        location.isMock
      } else {
        false
      }
}
